src/main/java/roomescape/domain/Reservation.java

```diff
@@ -0,0 +1,79 @@
+package roomescape.domain;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import roomescape.domain.exception.ReservationException;
+
+public final class Reservation {
+
+    private static final int MAX_NAME_LENGTH = 20;
+
+    private final Long id;
+    private final String name;
+    private final ReservationDate date;
+    private final ReservationTime time;
+
+    public Reservation(Long id, String name, LocalDateTime startDateTime) {
+        validateNameLength(name);
+        validateNotPastDateTime(startDateTime);
+        this.id = id;
+        this.name = name;
+        this.date = new ReservationDate(startDateTime.toLocalDate());
+        this.time = new ReservationTime(startDateTime.toLocalTime());
+    }
+
+    public Reservation(Long id, String name, LocalDate date, LocalTime time) {
+        this(id, name, LocalDateTime.of(date, time));
+    }
+
+    public Reservation(String name, LocalDate date, LocalTime time) {
+        this(null, name, LocalDateTime.of(date, time));
+    }
+
+    public Reservation(Long id, String name, ReservationDate reservationDate, ReservationTime reservationTime) {
+        validateNameLength(name);
+        validateNotPastDateTime(LocalDateTime.of(reservationDate.getStartDate(), reservationTime.getStartTime()));
+        this.id = id;
+        this.name = name;
+        this.date = reservationDate;
+        this.time = reservationTime;
+    }
+
+    public Reservation(String name, ReservationDate reservationDate, ReservationTime reservationTime) {
+        validateNameLength(name);
+        validateNotPastDateTime(LocalDateTime.of(reservationDate.getStartDate(), reservationTime.getStartTime()));
```

Comment:
```suggestion
    public Reservation(String name, ReservationDate reservationDate, ReservationTime reservationTime) {
        this(null, name // 추가
```
생성자마다 중복될 것 같아요
this를 통해 중복을 줄여볼까요?

Reply:
감사합니다! 덕분에 중복이 제거되고 코드가 더 깔끔해졌습니다. 👍

[refactor: 생성자에서 this() 활용하여 중복 제거](https://github.com/woowacourse/spring-roomescape-admin/pull/314/commits/90fcb96ac50246c4b9e109975b76707e445b9fe2)